src/librustc/ty/mod.rs

```diff
@@ -2693,13 +2677,31 @@ fn def_span<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Span {
     tcx.hir.span_if_local(def_id).unwrap()
 }
 
+/// If the given def ID describes an item belonging to a trait,
+/// return the ID of the trait that the trait item belongs to.
+/// Otherwise, return `None`.
+fn trait_of_item<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Option<DefId> {
+    if def_id.krate != LOCAL_CRATE {
+        return tcx.trait_of_item(def_id)
```

Comment:
I am pretty sure that this `if` can be removed -- or else there is a bug. After all, `tcx.trait_of_item(def_id)` should just call back into this code, since this is a query implementation.